Improvement: Bump errorprone to 2.11.0

robert3005 opened the pull request:
## Before this PR
<!-- What's wrong with the current state of the world and why change it now? -->

## After this PR
<!-- User-facing outcomes this PR delivers go below -->
==COMMIT_MSG==
Upgrade errorprone to 2.11.0
==COMMIT_MSG==

Since we bundle mockito errorprone rules and there was a slight api break in errorprone we need mockito to release their updated rules first https://github.com/mockito/mockito/pull/2553

## Possible downsides?
<!-- Please describe any way users could be negatively affected by this PR. -->

svc-excavator-bot committed:
Excavator: Upgrade dependencies

Commit:
opts

Commit:
Improvement: Bump errorprone to 2.11.0

robert3005 commented on baseline-error-prone/build.gradle:
```diff
@@ -36,14 +36,30 @@ dependencies {
 
+// Incorrectly identifies tests as junit4 usage
+tasks.checkJUnitDependencies.enabled = false
+tasks.javadoc.enabled = false
```
I have no idea how to pass `--add-exports` flags to javadoc in gradle (options are definitely not being respected) and at this point I have given up and disabled javadoc for errorprone.

robert3005 commented on build.gradle:
```diff
@@ -42,6 +42,7 @@ allprojects {
         tasks.withType(JavaCompile) {
             options.compilerArgs += ['-Werror', '-Xlint:deprecation']
             options.errorprone {
+                enabled = false
```
this can be reenabled once we release this pr and pick it up - otherwise we fail because of the break

ferozco commented on baseline-error-prone/build.gradle:
```diff
@@ -36,14 +36,30 @@ dependencies {
+        '--add-exports=jdk.compiler/com.sun.tools.javac.parser=ALL-UNNAMED',
+        '--add-exports=jdk.compiler/com.sun.tools.javac.processing=ALL-UNNAMED',
+        '--add-exports=jdk.compiler/com.sun.tools.javac.tree=ALL-UNNAMED',
+        '--add-exports=jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED'
```
should we just set this in `gradle.properties`? I think it will fix the java doc problem

robert3005 replied:
It does not, I spent couple hours trying to make it work and nothing made sense. Since with toolchains with fork the jvm for compilation add this to gradle.properties doesn't fix compilation

robert3005 commented on gradle-baseline-java/src/test/groovy/com/palantir/baseline/BaselineErrorProneRefasterIntegrationTest.groovy:
```diff
@@ -152,6 +152,10 @@ class BaselineErrorProneRefasterIntegrationTest extends AbstractPluginTest {
         when:
         buildFile << standardBuildFile
         buildFile << '''
+        tasks.withType(JavaCompile) {
```
this is unnecessary once mockito releases with bumped errorprone

robert3005 commented on gradle-baseline-java/src/test/groovy/com/palantir/baseline/BaselineExactDependenciesTest.groovy:
```diff
@@ -68,6 +68,9 @@ class BaselineExactDependenciesTest extends AbstractPluginTest {
                 mavenCentral()
                 mavenLocal() // for baseline-error-prone
             }
+            tasks.withType(JavaCompile) {
```
this is unnecessary once mockito releases with bumped errorprone

avisbal commented:
I'm seeing the following error in some excavators, would you mind updating `AvoidNewHashMapInt` as well? I'm guessing there was a break for `MethodMatchers$ConstructorClassMatcher.withParameters(java.lang.String[])`.

```
Caused by: java.lang.NoSuchMethodError: 'com.google.errorprone.matchers.method.MethodMatchers$ParameterMatcher com.google.errorprone.matchers.method.MethodMatchers$ConstructorClassMatcher.withParameters(java.lang.String[])'
	at com.palantir.baseline.errorprone.AvoidNewHashMapInt.<clinit>(AvoidNewHashMapInt.java:48)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at java.base/java.util.ServiceLoader$ProviderImpl.newInstance(ServiceLoader.java:780)
	... 49 more
```

robert3005 commented:
I already have - the problem is that you have the rule compiled against 2.10.0 and you try to run it against 2.11.0. What you're seeing is a linking problem - if you'd recompile the method against new errorprone it will be fine since signature is api but not abi compatible

pkoenig10 commented:
There nothing we can do for `AvoidNewHashMapInt` except doing this upgrade. The method signature was changed, so the new method does not existing in error-prone 2.10.0.
https://github.com/google/error-prone/commit/ccb01275000f26edc260f606e42d7528a692a5b7#diff-146fcbeaf981564e0494974aa93adbd8f68ffa92f25df4cda2f899a94d97e6d4

Commit:
Update

Commit:
opt out of mockito-error-prone

Commit:
Fix BugPatternNaming

```
./gradlew classes testClasses -PerrorProneApply=BugPatternNaming
./gradlew format
```

Commit:
drop mockito-error-prone

Commit:
remove test cruft

svc-autorelease commented:
Released [4.82.0](https://github.com/palantir/gradle-baseline/releases/tag/4.82.0)
